File: Programming-Basics-With-JavaScript/Homeworks-And-Labs/02.Conditional-Statements/03-additional/03.harvest.js
function harvest(input) {

    const grapesPerLiter = 2.5;

    const xArea = Number(input[0]);
    const yGrapes = Number(input[1]);
    const zNeededLiters = Number(input[2]);
    const workmen = Number(input[3]);

    const totalGrapes = xArea * yGrapes;
    const producedWine = totalGrapes / grapesPerLiter * 0.4;
    const difference = Math.abs(producedWine - zNeededLiters);

    if (producedWine < zNeededLiters) {
        console.log(`It will be a tough winter! More ${Math.floor(difference)} liters wine needed.`);
    } else {
        const extraWine = difference / workmen;
        console.log(`Good harvest this year! Total wine: ${Math.floor(producedWine)} liters.`);
        console.log(`${Math.ceil(difference)} liters left -> ${Math.ceil(extraWine)} liters per person.`);
    }
}
harvest(["650", "2", "175", "3"]);

File: Python_Basics/Homeworks-And-Labs/02.conditional_statements/03.additional/03.harvest.py
from math import ceil, floor

grapesPerLiter = 2.5

xArea = int(input())
yGrapes = float(input())
zNeededLiters = int(input())
workmen = int(input())

totalGrapes = xArea * yGrapes
producedWine = totalGrapes / grapesPerLiter * 0.4
difference = abs(producedWine - zNeededLiters)

if producedWine < zNeededLiters:
    print(f'It will be a tough winter! More {floor(difference)} liters wine needed.')
else:
    extraWine = difference / workmen
    print(f'Good harvest this year! Total wine: {floor(producedWine)} liters.')
    print(f'{ceil(difference)} liters left -> {ceil(extraWine)} liters per person.')
